tensorflow/core/kernels/fused_batch_norm_op.cc

```diff
@@ -676,6 +678,15 @@ bool BatchnormSpatialPersistentEnabled() {
     return is_enabled;
   }();
   return is_enabled;
+#elif TF_ROCM_VERSION >= 50000
+  static bool is_enabled = [] {
+    bool is_enabled = false;
+    TF_CHECK_OK(tensorflow::ReadBoolFromEnvVar(
+        "TF_USE_ROCM_NHWC",
+        /*default_val=*/false, &is_enabled));
+    return is_enabled;
+  }();
+  return is_enabled;
```

Comment:
Is this the correct place for this check?

What we really want is the `fast_nhwc_batch_norm` var to evaluate to `true` when this condition is met. The change above will work, but we seem to be hijacking the unrelated `BatchnormSpatialPersistentEnabled` routine for it.


How about we have something like
```
bool useNhwcLayoutForBatchNormOnRocm() {
#if TENSORFLOW_USE_ROCM && (TF_ROCM_VERSION >= 50000)
     // add above logic here...perhaps have separate env vars for conv and batchnorm
#else                 
   return false;
}

bool useNhwcLayoutForConvOnRocm() {
...
}
```
If we can stick them in a header file, which is visible to the layout optimizer, conv impl, and  batchnorm impl codes, then we call avoid the code duplication

That will also help get rid of the other `(TF_ROCM_VERSION >= 50000)` checks in this file